Extension installes wrong binaries (VSC-1006)

### OS

Linux

### Operating System version

Linux RPi2 5.15.76-v7+ #1597 SMP Fri Nov 4 12:13:17 GMT 2022 armv7l GNU/Linux
Linux RPi4 5.15.76-v7l+ #1597 SMP Fri Nov 4 12:14:58 GMT 2022 armv7l GNU/Linux

### Visual Studio Code version

1.73.1-1667960052

### ESP-IDF version

v4.4.3; release/v4.4; release/v5.0; master

### Python version

3.9.2

### Doctor command output

n/a

### Extension

1.5.1

### Description

i followed https://github.com/espressif/vscode-esp-idf-extension/blob/master/docs/tutorial/install.md

- my Raspberry Pi 2B and Raspberry Pi 4 B (4GByte) is with RaspOS Bullseye
- i installed VS Core 1.73.1 with `sudo apt install code`
- i installed all requirements with `sudo apt-get install git wget flex bison gperf python3 python3-venv cmake ninja-build ccache libffi-dev libssl-dev dfu-util libusb-1.0-0`
- i installed the Espressif IDF Extension 1.5.1.
- i run the `ESP-IDF:Setup (ESP-IDF: Configure Extension)`, installed ESP-IDF, ESP-IDF-Tools, python env.
- i copied the udev rule `60-openocd.rules` in place.
- i enabled to trust all the subfolders of the projects folder.

but when i run the `ESP-IDF:Setup (ESP-IDF: Configure Extension)`, it installes some wrong binaries for `amd64`, `x86_64` and `linux64` on my Raspberry Pi that only can handle `armhf` binaries.

it happens in the `Express` and `Advanced` setup mode.
i tried v4.4.3, release/v4.4, release/v5.0 and master of esp-idf.

i also tried to clone the esp-idf manually (`git clone --recursive https://github.com/espressif/esp-idf.git`) and installed the tools manually (`. ./install.sh`) to get the correct armhf binaries,

but when i run the `ESP-IDF:Setup (ESP-IDF: Configure Extension)` in advanced mode (with `Find ESP-IDF on your system`) and selected all the existing esp-idf `/home/admin/esp/esp-idf` and tools `/home/admin/.espressif` pathes from my manual clone and installation, the setup still begin to download some tools in the `amd64` version and overrides(?) the existing correct `armhf` ones or at least ignores the armhf ones and uses the wrong ones.

the result is the same - i can not config/build/flash/monitor any example in VS Core + Espressif IDF Extension.

for example, `cmake` i installed via `apt`, but VS Core tries to execute the `cmake` installed from the esp-idf in the tools path that is not `armfh` binaries -  `Downloading CMake build system; cmake-3.24.0-linux-x86_64.tar.gz has (45941) KB` - so it is `x86_64`

![ESP-IDF-example](https://user-images.githubusercontent.com/4750719/201492267-9bf2f752-c512-4a38-be48-218208ab3608.png)


### Debug Message

```plain
n/a
```


### Other Steps to Reproduce

_No response_

### I have checked existing issues, online documentation and the Troubleshooting Guide

- [X] I confirm I have checked existing issues, online documentation and Troubleshooting guide.

a little bit more progress.
i tried again manually. clone  esp-idf repository, install esp-idf-tools, exported the environment variables and started VS Code from that environment.
i keept all the already installed esp-idf-tools in ESP-IDF Extension; ADVANCED setup untouched.

now it starts building the `hello_wold` example - but it ends up  with errors because of missing module `idf_component_manager`.
i can sucessfully build the `hello_world` project from the same export-environment outside of VS Code from the console by using `idf.py build`.
or even worse, i can build the `hello_world` project inside VS Code, when i open a  terminal and type in `idf.py build`.


- i deleted all the mess from earlier tries:
  ```bash
  rm -rf ~/.espressif/
  rm -rf ~/esp/esp-idf/
  rm -rf ~/.vscode/
  rm -rf ~/.config/Code/
  ```
- i cloend v4.4.3:
  ```bash
  git clone --branch=v4.4.3 --recurse-submodules https://github.com/espressif/esp-idf ~/esp/esp-idf/
  ```
- i installed the esp-idf tools:
  ```bash
  cd ~/esp/esp-idf/
  . ./install.sh esp32
  ```
- i run code from console with exports:
  ```bash
  cd ~/esp/esp-idf/
  . ./export.sh
  code
  ```
- by starting VS Code from that export-environment:
  - i installed the `Espressif IDF Extension`.
    ![02](https://user-images.githubusercontent.com/4750719/201522440-96a941ef-69d0-47d5-88b6-4bbc042ed42f.png)
  - in `ESP-IDF: Configure ESP-IDF extension`, i choosed `ADVANCED` setup.
    ![03](https://user-images.githubusercontent.com/4750719/201522506-db414d22-f5ef-4210-ba4c-bb67f6e061bd.png)
    - selected "Find ESP-IDF in your system", ...
      ![04](https://user-images.githubusercontent.com/4750719/201522518-4dea2a89-6247-4e86-a35e-ac1bbc5e8ba6.png)
    - selected "Use existing ESP-IDF Tools"
      ![05](https://user-images.githubusercontent.com/4750719/201522528-bae924d1-bebf-4846-8935-7d421d6112f2.png)
    - copied `60-openocd.rules` in to place
      ![06](https://user-images.githubusercontent.com/4750719/201522537-8dc13fa2-ccf6-4747-b167-e14de948bfdc.png)
  - i opend `ESP-IDF: Show Examples Projects` and created the hello_world project to my projects folder
    ![07](https://user-images.githubusercontent.com/4750719/201522844-43a33d96-8e0b-4d21-9f17-8aaace74aaed.png)
  - closed VS Code
- by starting VS Code from that export-environment the second time:
  - i enabled to trust all files and parent folders of my projects folder
    ![08](https://user-images.githubusercontent.com/4750719/201522927-2b356715-82e7-47ad-81ce-a9c959fb0753.png)
  - and tried to build the `hello_world` project - it ends up with errors:
    ![09](https://user-images.githubusercontent.com/4750719/201522947-352db3c4-c8db-4aa8-84c0-859862e4d00e.png)
  - but when i open a terminal in VS Code and run idf.py build, it builds with success.
    ![10](https://user-images.githubusercontent.com/4750719/201523410-1c7486f6-b5eb-4170-a844-ed4de5d2e631.png)
    ![11](https://user-images.githubusercontent.com/4750719/201523422-ee1a19ce-3cd9-4a7f-a50e-6304cd8a7e2d.png)


here the log text:
```log
 *  Executing task: cmake -G Ninja -DPYTHON_DEPS_CHECKED=1 -DESP_PLATFORM=1 -B /media/pi/data/projects/esp32/hello_world/build -S /media/pi/data/projects/esp32/hello_world 

-- Found Git: /usr/bin/git (found version "2.30.2") 
-- The C compiler identification is GNU 8.4.0
-- The CXX compiler identification is GNU 8.4.0
-- The ASM compiler identification is GNU
-- Found assembler: /home/pi/.espressif/tools/xtensa-esp32-elf/esp-2021r2-patch5-8.4.0/xtensa-esp32-elf/bin/xtensa-esp32-elf-gcc
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /home/pi/.espressif/tools/xtensa-esp32-elf/esp-2021r2-patch5-8.4.0/xtensa-esp32-elf/bin/xtensa-esp32-elf-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /home/pi/.espressif/tools/xtensa-esp32-elf/esp-2021r2-patch5-8.4.0/xtensa-esp32-elf/bin/xtensa-esp32-elf-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Project is not inside a git repository, or git repository has no commits; will not use 'git describe' to determine PROJECT_VER.
-- Building ESP-IDF components for target esp32
CMake Error at /home/pi/esp/esp-idf/tools/cmake/build.cmake:452 (message):
  /usr/bin/python: Error while finding module specification for
  'idf_component_manager.prepare_components' (ModuleNotFoundError: No module
  named 'idf_component_manager')

Call Stack (most recent call first):
  /home/pi/esp/esp-idf/tools/cmake/project.cmake:384 (idf_build_process)
  CMakeLists.txt:6 (project)


-- Configuring incomplete, errors occurred!
See also "/media/pi/data/projects/esp32/hello_world/build/CMakeFiles/CMakeOutput.log".

 *  The terminal process "/usr/bin/bash '-c', 'cmake -G Ninja -DPYTHON_DEPS_CHECKED=1 -DESP_PLATFORM=1 -B /media/pi/data/projects/esp32/hello_world/build -S /media/pi/data/projects/esp32/hello_world'" terminated with exit code: 1. 
```

--- a/src/PlatformInformation.ts
+++ b/src/PlatformInformation.ts
@@ -40,7 +40,7 @@ export class PlatformInformation {
   public get platformToUse(): string {
     switch (this.platform) {
       case "darwin":
-        return this.architecture === "arm64" ? "macos-arm64": "macos";
+        return this.architecture === "arm64" ? "macos-arm64" : "macos";
       case "win32":
         return this.architecture === "x86" ? "win32" : "win64";
       case "linux":

--- a/src/PlatformInformation.ts
+++ b/src/PlatformInformation.ts
@@ -40,14 +40,17 @@ export class PlatformInformation {
   public get platformToUse(): string {
     switch (this.platform) {
       case "darwin":
-        return "macos";
+        return this.architecture === "arm64" ? "macos-arm64": "macos";
       case "win32":
         return this.architecture === "x86" ? "win32" : "win64";
       case "linux":
         switch (this.architecture) {
           case "arm64":
+          case "armv8l":
+          case "aarch64":
             return "linux-arm64";
           case "armel":
+          case "armv7l":
             return "linux-armel";
           case "armhf":
             return "linux-armhf";

--- a/src/setup/setupInit.ts
+++ b/src/setup/setupInit.ts
@@ -318,7 +318,13 @@ export async function getSetupInitialValues(
         prevInstall.toolsResults
       );
 
-      if (cmakeFromToolsIndex !== -1) {
+      const canAccessCMake = await utils.isBinInPath(
+        "cmake",
+        extensionPath,
+        process.env
+      );
+
+      if (cmakeFromToolsIndex !== -1 || canAccessCMake) {
         prevInstall.toolsResults.splice(cmakeFromToolsIndex, 1);
       } else {
         setupInitArgs.onReqPkgs = setupInitArgs.onReqPkgs
@@ -331,7 +337,13 @@ export async function getSetupInitialValues(
         prevInstall.toolsResults
       );
 
-      if (ninjaFromToolsIndex !== -1) {
+      const canAccessNinja = await utils.isBinInPath(
+        "ninja",
+        extensionPath,
+        process.env
+      );
+
+      if (ninjaFromToolsIndex !== -1 || canAccessNinja) {
         prevInstall.toolsResults.splice(ninjaFromToolsIndex, 1);
       } else {
         setupInitArgs.onReqPkgs = setupInitArgs.onReqPkgs

--- a/src/setup/setupInit.ts
+++ b/src/setup/setupInit.ts
@@ -324,9 +324,9 @@ export async function getSetupInitialValues(
         process.env
       );
 
-      if (cmakeFromToolsIndex !== -1 || canAccessCMake) {
+      if (cmakeFromToolsIndex !== -1) {
         prevInstall.toolsResults.splice(cmakeFromToolsIndex, 1);
-      } else {
+      } else if (canAccessCMake === "") {
         setupInitArgs.onReqPkgs = setupInitArgs.onReqPkgs
           ? [...setupInitArgs.onReqPkgs, "cmake"]
           : ["cmake"];
@@ -343,9 +343,9 @@ export async function getSetupInitialValues(
         process.env
       );
 
-      if (ninjaFromToolsIndex !== -1 || canAccessNinja) {
+      if (ninjaFromToolsIndex !== -1) {
         prevInstall.toolsResults.splice(ninjaFromToolsIndex, 1);
-      } else {
+      } else if (canAccessNinja === "") {
         setupInitArgs.onReqPkgs = setupInitArgs.onReqPkgs
           ? [...setupInitArgs.onReqPkgs, "ninja"]
           : ["ninja"];

